hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Constants.java

```diff
@@ -778,18 +778,18 @@ private Constants() {
   /**
    * List of custom headers to be set on the service client.
    * Multiple parameters can be used to specify custom headers.
-   * fs.s3a.s3.custom.headers - headers to add on all the s3 requests.
-   * fs.s3a.sts.custom.headers - headers to add on all the sts requests.
+   * fs.s3a.client.s3.custom.headers - headers to add on all the s3 requests.
+   * fs.s3a.client.sts.custom.headers - headers to add on all the sts requests.
    * Examples
    * CustomHeader {@literal ->} 'Header1:Value1'
    * CustomHeaders {@literal ->} 'Header1=Value1:Value2,Header2=Value1'
    */
   public static final String CUSTOM_HEADERS_STS =
-      "fs.s3a." + Constants.AWS_SERVICE_IDENTIFIER_STS.toLowerCase()
+      "fs.s3a.client." + Constants.AWS_SERVICE_IDENTIFIER_STS.toLowerCase()
```

Comment:
I'd pull this out as something like CUSTOM_HEADER_PREFIX

Reply:
do you mean something like:
```
public static final String FS_S3A_CUSTOM_HEADER_PREFIX = "fs.s3a.client.";
public static final String FS_S3A_CUSTOM_HEADER_POSTFIX = ".custom.headers";

public static final String CUSTOM_HEADERS_STS = FS_S3A_CUSTOM_HEADER_PREFIX + Constants.AWS_SERVICE_IDENTIFIER_STS.toLowerCase() +
FS_S3A_CUSTOM_HEADER_POSTFIX;
```

Reply:
yes, but without the double ".." that would produce

Reply:
That would not right? `Constants.AWS_SERVICE_IDENTIFIER_STS.toLowerCase()` is just STS.